pyproject.toml

```diff
@@ -34,9 +34,8 @@ docstring-convention = "google"
 [tool.pyright]
 exclude = ["ops/_vendor", ".git", "__pycache__", ".tox", "build", "test",
     "ops/lib", "ops/_private", "dist", "*.egg_info", "venv", "ops/charm.py",
-    "ops/framework.py", "ops/jujuversion.py", "ops/log.py", "ops/main.py",
-    "ops/model.py", "ops/pebble.py", "ops/testing.py",
-    "ops/version.py", "setup.py", "ops/__init__.py"]
-pythonVersion = 3.5 # check no python > 3.5 features are used
+    "ops/framework.py", "ops/main.py", "ops/model.py", "ops/pebble.py",
+    "ops/testing.py", "ops/__init__.py"]
+pythonVersion = "3.5" # check no python > 3.5 features are used
```

Comment:
The analogous for mypy, `mypy --python-version 3.5`, is not sufficient (gives false negatives): we had to [run with py3.5](https://github.com/canonical/prometheus-k8s-operator/blob/6f2676c2674915777a694d8faa11df389a1b5670/.github/workflows/pull-request.yaml#L34).
Idk if pyright has the same issue.

Reply:
mmm. Let's add that just to be sure.

Reply:
And that's how I discover that pyright supports python3.7+

Reply:
Not sure if it's worth to support both mypy and pyright at this point, and use mypy for python 3.5. It looks like a lot more work.